returnn/forward.py

```diff
@@ -0,0 +1,213 @@
+__all__ = ["ReturnnForwardJob"]
+
+from sisyphus import *
+
+import copy
+import glob
+import os
+import shutil
+import subprocess as sp
+import tempfile
+
+from i6_core.returnn.config import ReturnnConfig
+from i6_core.returnn.training import Checkpoint
+import i6_core.util as util
+
+Path = setup_path(__package__)
+
+
+class ReturnnForwardJob(Job):
+    """
+    Run a RETURNN "forward" pass to HDF with a specified model checkpoint.
+    Also allows to run an "eval" task pass, which is similar to "forward" but treats all layers as in training mode,
+    which can be used to e.g. do cheating experiments.
+
+    Outputs:
+
+    Dict[tk.Path] out_hdf_files: Dictionary of all output HDF files that were requested by the key list `hdf_outputs`
+    tk.Path out_default_hdf: For forward (not eval) mode, this contains the default HDF file that is always written
+        by RETURNN, independent of other settings.
+    """
+
+    def __init__(
+        self,
+        model_checkpoint,
+        returnn_config,
+        hdf_outputs=None,
+        eval_mode=False,
+        *,  # args below are keyword only
+        log_verbosity=3,
+        device="gpu",
+        time_rqmt=4,
+        mem_rqmt=4,
+        cpu_rqmt=4,
```

Comment:
is this a reasonable default for a job that does not do CPU heavy load? 
```suggestion
        cpu_rqmt=1,
```